include/xtensor-io/xfile_array.hpp

```diff
@@ -8,6 +8,11 @@
 #include <xtensor/xarray.hpp>
 #include <xtensor/xnoalias.hpp>
 
+#define XFILE_ARRAY_NOT_DIRTY   0
+#define XFILE_ARRAY_ALL_DIRTY   0xFF
+#define XFILE_ARRAY_DATA_DIRTY  1
+#define XFILE_ARRAY_SHAPE_DIRTY (1 << 1)
```

Comment:
An enum class in the xt namespace would be better than polluting the global namespace with C-style macros.
But I actually think that a simple struct with `shape_dirty` and `data_dirty` members would be actually easier to read and understand than a single integer and boolean masking.

Reply:
:+1: